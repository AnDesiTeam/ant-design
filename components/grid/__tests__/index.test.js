import React from 'react';
import { render, mount } from 'enzyme';
import { Col, Row } from '..';
import mountTest from '../../../tests/shared/mountTest';
import rtlTest from '../../../tests/shared/rtlTest';
import useBreakpoint from '../hooks/useBreakpoint';

describe('Grid', () => {
  mountTest(Row);
  mountTest(Col);

  rtlTest(Row);
  rtlTest(Col);

  it('should render Col', () => {
    const wrapper = render(<Col span={2} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Row', () => {
    const wrapper = render(<Row />);
    expect(wrapper).toMatchSnapshot();
  });

  it('when typeof gutter is object', () => {
    const wrapper = mount(<Row gutter={{ xs: 8, sm: 16, md: 24 }} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').prop('style')).toEqual({
      marginLeft: -4,
      marginRight: -4,
    });
  });

  it('when typeof gutter is object array', () => {
    const wrapper = mount(
      <Row
        gutter={[
          { xs: 8, sm: 16, md: 24, lg: 32, xl: 40 },
          { xs: 8, sm: 16, md: 24, lg: 32, xl: 40 },
        ]}
      />,
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').prop('style')).toEqual({
      marginTop: -4,
      marginRight: -4,
      marginBottom: 4,
      marginLeft: -4,
    });
  });

  it('renders wrapped Col correctly', () => {
    const MyCol = () => <Col span={12} />;
    const wrapper = render(
      <Row gutter={20}>
        <div>
          <Col span={12} />
        </div>
        <MyCol />
      </Row>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should work correct when gutter is object', () => {
    const wrapper = mount(<Row gutter={{ xs: 20 }} />);
    expect(wrapper.find('div').prop('style')).toEqual({
      marginLeft: -10,
      marginRight: -10,
    });
  });

  it('should work currect when gutter is array', () => {
    const wrapper = mount(<Row gutter={[16, 20]} />);
    expect(wrapper.find('div').prop('style')).toEqual({
      marginLeft: -8,
      marginRight: -8,
      marginTop: -10,
      marginBottom: 10,
    });
  });

  // By jsdom mock, actual jsdom not implemented matchMedia
  // https://jestjs.io/docs/en/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
  it('should work with useBreakpoint', () => {
    function Demo() {
      const screens = useBreakpoint();

      return JSON.stringify(screens);
    }
    const wrapper = mount(<Demo />);

    expect(wrapper.text()).toEqual(
      JSON.stringify({
        xs: true,
        sm: false,
        md: false,
        lg: false,
        xl: false,
        xxl: false,
      }),
    );
  });
});
