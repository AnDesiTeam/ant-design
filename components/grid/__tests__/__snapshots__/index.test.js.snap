// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid renders wrapped Col correctly 1`] = `
<div
  class="ant-row"
  style="margin-left:-10px;margin-right:-10px"
>
  <div>
    <div
      class="ant-col ant-col-12"
      style="padding-left:10px;padding-right:10px"
    />
  </div>
  <div
    class="ant-col ant-col-12"
    style="padding-left:10px;padding-right:10px"
  />
</div>
`;

exports[`Grid rtl render component should be rendered correctly in RTL direction 1`] = `
<div
  class="ant-row ant-row-rtl"
/>
`;

exports[`Grid rtl render component should be rendered correctly in RTL direction 2`] = `
<div
  class="ant-col ant-col-rtl"
/>
`;

exports[`Grid should render Col 1`] = `
<div
  class="ant-col ant-col-2"
/>
`;

exports[`Grid should render Row 1`] = `
<div
  class="ant-row"
/>
`;

exports[`Grid when typeof gutter is object 1`] = `
<Row
  gutter={
    Object {
      "md": 24,
      "sm": 16,
      "xs": 8,
    }
  }
>
  <div
    className="ant-row"
    style={
      Object {
        "marginLeft": -4,
        "marginRight": -4,
      }
    }
  />
</Row>
`;

exports[`Grid when typeof gutter is object array 1`] = `
<Row
  gutter={
    Array [
      Object {
        "lg": 32,
        "md": 24,
        "sm": 16,
        "xl": 40,
        "xs": 8,
      },
      Object {
        "lg": 32,
        "md": 24,
        "sm": 16,
        "xl": 40,
        "xs": 8,
      },
    ]
  }
>
  <div
    className="ant-row"
    style={
      Object {
        "marginBottom": 4,
        "marginLeft": -4,
        "marginRight": -4,
        "marginTop": -4,
      }
    }
  />
</Row>
`;
